contributors/design-proposals/auth/svcacct-token-volume-source.md

```diff
@@ -0,0 +1,138 @@
+distribution with support for [improved service account
+tokens](https://github.com/kubernetes/community/pull/1460) and explores how to
+migrate from the existing mechanism backwards compatibly.
+
+## Motivation
+
+Many workloads running on Kubernetes need to prove to external parties who they
+are in order to participate in a larger application environment. This identity
+must be attested to by the orchestration system in a way that allows a third
+party to trust that an arbitrary container on the cluster is who it says it is.
+In addition, infrastructure running on top of Kubernetes needs a simple
+mechanism to communicate with the Kubernetes APIs and to provide more complex
+tooling. Finally, a significant set of security challenges are associated with
+storing service account tokens as secrets in Kubernetes and limiting the methods
+whereby malicious parties can get access to these tokens will reduce the risk of
+platform compromise.
+
+As a platform, Kubernetes should evolve to allow identity management systems to
+provide more powerful workload identity without breaking existing use cases, and
+provide a simple out of the box workload identity that is sufficient to cover
+the requirements of bootstrapping low-level infrastructure running on
+Kubernetes. We expect that other systems to cover the more advanced scenarios,
+and see this effort as necessary glue to allow more powerful systems to succeed.
+
+With this feature, we hope to provide a backwards compatible replacement for
+service account tokens that strengthens the security and improves the
+scalability of the platform.
+
+## Proposal
+
+Kubernetes should:
+
+1.  Allow the automatic service account token injection to be disabled via
+    configuration.
+1.  Allow automatic service account token creation to occur only when a user
+    explicitly requests.
+1.  Implement a ServiceAccountToken volume projection that that maintains a
+    service account token requested by the node from the TokenRequest API.
+
+### Token Volume Projection
+
+A new volume projection will be implemented with an API that closely matches the
+TokenRequest API.
+
+```go
+type ServiceAccountTokenProjection struct {
+  // Audiences are the intendend audiences of the token.
+  Audiences []string
+  // ExpirationSeconds is the requested duration of validity of the service
+  // account token
+  ExpirationSeconds *int64
+  // Path is the relative path of the file to project the token into.
+  Path string
+}
+```
+
+A volume plugin implemented in the kubelet will project a service account token
+sourced from the TokenReview API into volumes created from
+ProjectedVolumeSources. As the token approaches expiration, the kubelet volume
+plugin will proactively rotate the service account token.
+
+To replace the current service account token secrets, we also need to inject the
+clusters CA certificate bundle. Initially we will deploy to data in a configmap
```

Comment:
Having a way to *just* take the cluster CA bundle would be nice. This would allow a service to receive jwts from other services without having a SA token itself.

Reply:
Ah -- ignore -- thought this was part of the SA token projection..